### PR TITLE
fix: insert condition and mobile date picker

### PR DESCRIPTION
**Pull request by leleOFA**
## Short description
fix date picker when display is in landscape

## List of changes proposed in this pull request
- insert condition for mobile 
- insert component mobile date picker

## How to test
go to the notification page, choose filter with date and show dialog for choosing date

sudo yarn test CustomDatePicker.test.tsx

### CODE REVIEW

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
+/* eslint-disable sonarjs/no-identical-functions */
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:1]: Use https://mui.com/x/react-date-pickers/date-picker/ that automatically switch between the two views

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 import { de, enGB, fr, it, sl } from 'date-fns/locale';
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:2]: Create two test cases that check the two views

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 1 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx:1]: done

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 2 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx:2]: done

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 3]
```diff
@@ -26,6 +26,16 @@ describe('test CustomDatePicker component', () => {
   });
 });
 
+describe('test CustomDatePicker component in mobile', () => {
+  it('renders the component with mobile window', () => {
+    window.matchMedia = createMatchMedia(800);
+    const { getByPlaceholderText, container } = render(<RenderDatePicker />);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:32]: In this test we are not testing if mobile view is shown, because we are olny testing that an input is shown without testing the popup.

Furthermore, there is no test that tests the desktop view

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 4]
```diff
@@ -19,15 +16,17 @@ const locales: { [key: string]: Locale } = {
...
 const CustomDatePicker = (
-  props: DesktopDatePickerProps<DatePickerTypes> &
+  props: DatePickerProps<DatePickerTypes> &
     React.RefAttributes<HTMLDivElement> & { language?: string }
 ) => {
   const language = props.language ? props.language : 'it';
   return (
     <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={locales[`${language}`]}>
-      <DesktopDatePicker
+      <DatePicker
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:25]: The view on mobile devices is not good for our needs. Datepicker popup has a toolbar on top and has actions on bottom. You can remove it with slotProps https://mui.com/x/react-date-pickers/custom-layout/.

TIPS: define property slotProps in witch you remove toolbar and action bar. pay attention of the slotProps defined from the outside. you have to merge the slotProps defined here whit those from the outside. To hide the action bar you must pass empty actions array to the slotProps.

On the mobile view, the datepicker doesn't close on selection. Check on https://mui.com/x/api/date-pickers/date-picker/ how to do that.

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 5]
```diff
@@ -19,15 +16,17 @@ const locales: { [key: string]: Locale } = {
...
         {...props}
+        desktopModeMediaQuery="@media (pointer: fine)"
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:27]: remove this property because it is the default value

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 6]
```diff
@@ -19,15 +16,17 @@ const locales: { [key: string]: Locale } = {
...
         localeText={{
+          toolbarTitle: '',
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:29]: remove this property

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 5 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx:27]: done

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 6 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx:28]: done

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 4 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx:25]: done

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx [hunk 7]
```diff
@@ -189,8 +189,8 @@ const FilterNotifications = forwardRef(({ showFilters }: Props, ref) => {
     </CustomMobileDialog>
   ) : (
     <form onSubmit={formik.handleSubmit} data-testid="filter-form">
-      <Box
-        display={'flex'}
+      <Stack
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx:192]: Why this change?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx [hunk 8]
```diff
@@ -105,7 +104,8 @@ const FilterNotificationsFormBody = ({
         }
         size="small"
         fullWidth={isMobile}
-        sx={{ marginBottom: isMobile ? '20px' : '0' }}
+        sx={{ mb: isMobile ? '20px' : 0 }}
+        inputProps={{ maxLength: 25 }}
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:108]: why this change?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx [hunk 9]
```diff
@@ -122,65 +122,63 @@ const FilterNotificationsFormBody = ({
...
-        slotProps={{
-          textField: {
-            id: 'startDate',
-            name: 'startDate',
-            size: 'small',
-            inputProps: {
-              inputMode: 'text',
-              type: 'text',
-              'aria-label': t('filters.data_da-input-aria-label'),
+      <Stack direction={{ sm: 'column', xs: 'column' }}>
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:127]: why this change?

File: packages/pn-commons/src/test-utils.tsx [hunk 10]
```diff
@@ -88,9 +88,9 @@ const customRender = (
 
 // utility function
 /** This function simulate media query and is useful to test differences between mobile and desktop view */
-function createMatchMedia(width: number) {
+function createMatchMedia(width: number, height?: number) {
   return (query: string): MediaQueryList => ({
-    matches: mediaQuery.match(query, { width }),
+    matches: mediaQuery.match(query, { width, height }),
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/test-utils.tsx:93]: why the use of height?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 11]
```diff
@@ -51,11 +60,15 @@ describe('test CustomDatePicker languages', () => {
   ];
 
   it.each(languages)('check january month to be $month in $language', (language) => {
-    const { container, getByRole } = render(<RenderDatePicker language={language.language} />);
+    const { container, getByLabelText } = render(<RenderDatePicker language={language.language} />);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:63]: why the use of getByLabelText instead of getByRole?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 12]
```diff
@@ -51,11 +60,15 @@ describe('test CustomDatePicker languages', () => {
...
     expect(container).toHaveTextContent(/datepicker/i);
-    const button = getByRole('button');
+    const button = getByLabelText(/DatePicker/i);
     fireEvent.click(button);
-    const regExMonth = new RegExp(`${language.month}`, 'i');
-    expect(screen.getByText(regExMonth)).toBeInTheDocument();
+    const regExMonth = new RegExp(`${language.month}`);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:67]: why this change?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 13]
```diff
@@ -34,15 +43,15 @@ describe('test CustomDatePicker languages', () => {
     },
     {
       language: 'en',
-      month: 'january',
+      month: 'January',
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:46]: Why this change?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 14]
```diff
@@ -26,6 +26,16 @@ describe('test CustomDatePicker component', () => {
...
+  it('renders the component with mobile window', () => {
+    window.matchMedia = createMatchMedia(800);
+    const { getByPlaceholderText, container } = render(<RenderDatePicker />);
+    const input = getByPlaceholderText(/datepickerinput/i);
+    expect(container).toHaveTextContent(/datepicker/i);
+    expect(input).toBeInTheDocument();
+  });
+});
+
 describe('test CustomDatePicker languages', () => {
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:38]: this wasn't done

File: packages/pn-commons/src/components/CustomDatePicker.tsx [hunk 15]
```diff
@@ -19,14 +16,21 @@ const locales: { [key: string]: Locale } = {
...
   const language = props.language ? props.language : 'it';
   return (
     <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={locales[`${language}`]}>
-      <DesktopDatePicker
+      <DatePicker
         {...props}
+        slotProps={{
+          ...props.slotProps,
+          toolbar: { hidden: true },
+          textField: { size: 'small' },
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CustomDatePicker.tsx:30]: this property shouldn't be set

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 16]
```diff
@@ -19,11 +19,20 @@ const RenderDatePicker = ({ language = 'it' }: { language?: string }) => (
 
 describe('test CustomDatePicker component', () => {
   it('renders the component', () => {
-    const { getByPlaceholderText, container } = render(<RenderDatePicker />);
-    const input = getByPlaceholderText(/datepickerinput/i);
+    const { getByLabelText, container } = render(<RenderDatePicker />);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:22]: why this change?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 17]
```diff
@@ -26,6 +26,16 @@ describe('test CustomDatePicker component', () => {
   });
 });
 
+describe('test CustomDatePicker component in mobile', () => {
+  it('renders the component with mobile window', () => {
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:28]: this wasn't done

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 8 above)
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:108]: isn't it the same?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx (hunk 7 above)
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx:192]: isn't it the same?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 9 above)
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx]: fixed

File: packages/pn-commons/src/test-utils.tsx (hunk 10 above)
Comment (leleOFA) [reply packages/pn-commons/src/test-utils.tsx:93]: for emulate in test small height and also because it could be useful to us

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 11 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:63]: the previous test failure because not found role

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 3 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx]: this wasn't done

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 12 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:67]: same as before commnet

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx (hunk 7 above)
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx:192]: do you have checked all the layouts and all the use cases?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 13 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:46]: in hmtl I find the large letter in some months, so I adapted it to html

File: packages/pn-commons/src/components/CustomDatePicker.tsx (hunk 15 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/CustomDatePicker.tsx]: fixesd

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx [hunk 18]
```diff
@@ -168,16 +166,16 @@ const FilterNotificationsFormBody = ({
...
               inputMode: 'text',
               type: 'text',
               'aria-label': t('filters.data_a-input-aria-label'),
             },
+            size: 'small',
             fullWidth: isMobile,
-            sx: { marginBottom: isMobile ? '20px' : '0' },
           },
         }}
+        sx={{ mb: isMobile ? '20px' : 0 }}
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:178]: isn't it the same?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx [hunk 19]
```diff
@@ -140,16 +138,16 @@ const FilterNotificationsFormBody = ({
...
               inputMode: 'text',
               type: 'text',
               'aria-label': t('filters.data_da-input-aria-label'),
             },
+            size: 'small',
             fullWidth: isMobile,
-            sx: { marginBottom: isMobile ? '20px' : '0' },
           },
         }}
+        sx={{ mb: isMobile ? '20px' : 0 }}
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:150]: isn't it the same?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 16 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:22]: new component not modify placeholder and i use label

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 12 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:67]: can you copy the comment here? i don't find the comment

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 11 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:63]: maybe fails because by default it renders mobile view and the mobile view doesn't have the button. revert the changes and force the desktop view using createMatchMedia

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 12 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:67]: same as before

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 13 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:46]: with lowercase as before the test fails?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 16 above)
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:22]: i don't understand. texfield in RenderDatePicker has placeholder set to placeholder. why are you saying that there is no placeholder?

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 18 above)
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:178]: you put the sx outside the inputProps

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:150]: you put the sx outside the inputProps

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx (hunk 7 above)
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotifications.tsx:192]: yes, all works

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 8 above)
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:108]: delete it

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 11 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:63]: how do i emulate without using the specific hook?

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 13 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:46]: yes, in fact, the tests that failed before were precisely these

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 18 above)
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:178]: I thought that passing it into the component in slot props could create problems

File: packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-pa-webapp/src/components/Notifications/FilterNotificationsFormBody.tsx:150]: I thought that passing it into the component in slot props could create problems

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 16 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:22]: it fails to take the placeholder that I insert in all tests

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 20]
```diff
@@ -19,36 +42,16 @@ const RenderDatePicker = ({ language = 'it' }: { language?: string }) => (
 
 describe('test CustomDatePicker component', () => {
   it('renders the component', () => {
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:44]: ues async here

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 21]
```diff
@@ -19,36 +42,16 @@ const RenderDatePicker = ({ language = 'it' }: { language?: string }) => (
...
-    const { getByPlaceholderText, container } = render(<RenderDatePicker />);
-    const input = getByPlaceholderText(/datepickerinput/i);
-    expect(container).toHaveTextContent(/datepicker/i);
+    const { getByTestId, container } = render(<RenderDatePicker />);
+    const input = getByTestId(/calendaricon/i);
     expect(input).toBeInTheDocument();
+    fireEvent.click(input);
+    waitFor(async () => {
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:49]: change this with
```
const datepicker = await waitFor(() => getByRole('presentation'));
expect(datePicker).toBeInTheDocument();
```

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx [hunk 22]
```diff
@@ -19,36 +42,16 @@ const RenderDatePicker = ({ language = 'it' }: { language?: string }) => (
...
+      const datePicker = await screen.findByRole('presentation');
+      expect(datePicker).toBeInTheDocument();
+    });
+    expect(container).toHaveTextContent(/datepicker/i);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:53]: move this up to fireEvent

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 20 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx]: done

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 21 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx]: done

File: packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx (hunk 22 above)
Comment (leleOFA) [reply packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx:46]: done

**Review by AndreaCimini90 — APPROVED** [review]
